### ISSUE TITLE
DIY lenticular autostereoscopic 3D screens, and automatic pixel adjustment to overcome parallax issues.

### ISSUE THREAD

**Issue by gjupp** [vpinball/vpinball#999]
**Is your feature request related to a problem? Please describe.**
3D screens are difficult to come by, especially 4K at under 49" which is very large for a cabinet. Perhaps it may be possible to build one using a lenticular lens over the top of a non 3D screen. 42" 1080p passive 3D screens are easier to find secondhand but discontinued.

**Describe the solution you'd like**
I have been thinking about the use of lenticular lenses for creating DIY autostereoscopic 3D screens for pincabs, basically attaching a lenticular lens directly to the outer screen surface or even inbetween the outer and inner screen layers. I'm wondering if it might be possible if VPX could auto calculate (in realtime) the variable pixel alignment that would be needed to prevent parallax errors from the player's viewpoint. You could use headtracking to determine the calculation based on screen size/angle/player position/resolution/lenticular lens pitch/distance of lens from actual panel surface. You would also likely need a way to fine tune manually.

Without auto real time adjustment there would likely be a lot of parallax problems when viewed from a pinball cab player position at close range and there would likely only be a sweet spot where it looked 3D but the rest of the screen would look a bit garbled due to parallax error.

From what I have read the autostereoscopic screens are starting to mature in tech but more likely to be used for landscape orientation. There are vertical/portrait screens but these seem too large for pinball cab. Some of the screens already use head tracking to change the perspective of the image from the viewers POV.

Imagine if it were possible to create a autostereoscopic 3D screen just by applying a relatively low cost lens to your existing screen.

There may be issues that need to be overcome in relation to how the screen displays in 2D application and perhaps the idea is a bit naive in what is really required. Overall perceived resolution would likely be reduced by at least half on one axis similar to how LG FPR 3D loses resolution.

Maybe in the future we'll see some consumer autostereoscopic screens of the size required for pinball, but the question will be whether they will work at close range at an angle or in portrait at all.

Thoughts?

**Comment by Pittyh** [comment]
Imagine sticking on a $10 sheet of plastic on your cab screen for instant 3D :D

**Comment by gjupp** [comment]
Maybe not quite that cheap for a 42"-43" diagonal sheet but yeah possibly a cheapish solution. You might need to order a custom size and custom pitch to match the pitch of the screen for maximum resolution. That might involve needing to bulk order a bunch of lenses to fit say an LG C2-C3 42" OLED rather than one-off. Precision would also be key as if the pitch was off by even a tiny amount compared to the screen that would likely cause problems, you may be able to account for this kind of issue in software though.

**Comment by gjupp** [comment]
I was speaking with Blueskydefender (Depth3D coder) on discord and mentioned this idea to him. To my surprise he told me that a previous version of his Reshade Depth3D shader had an Autostereoscopic option but he removed it as it it was difficult to set up and only worked with one particular screen. He said his implementation of the headtracking use Freepie and Tobii eyetracking.

**Comment by pacleary** [comment]
I thought about this roughly the same time you posted it. It is a very niche request considering the logistics to acquiring this type of lens at the correct sizes, but I'm curious if it could possibly work.

**Comment by gjupp** [comment]
> I thought about this roughly the same time you posted it. It is a very niche request considering the logistics to acquiring this type of lens at the correct sizes, but I'm curious if it could possibly work.

Yeah, would likely need a custom lens manufacturer willing to make lenses to match the pitch of certain screens or one particular screen at least. Let's say it was an LG C2 or C3 for instance, and all owners of that screen wanting to participate could put in for the bulk buy.
